src/app/components/update-itinerary/update-itinerary.component.ts

```diff
@@ -0,0 +1,135 @@
+import {Component, OnInit} from '@angular/core';
+import {AbstractControl, FormBuilder, FormGroup, UntypedFormGroup, Validators} from "@angular/forms";
+import {ItenaryService} from "../../services/itinerary.service";
+import {ActivatedRoute, Router} from "@angular/router";
+import {Itinerary} from "../../models/itinenary";
+
+@Component({
+  selector: 'app-update-itinerary',
+  templateUrl: './update-itinerary.component.html',
+  styleUrls: ['./update-itinerary.component.scss']
+})
+export class UpdateItineraryComponent implements OnInit{
+  validateForm!: UntypedFormGroup;
+  submitForm(): void {
+    if (this.validateForm.valid) {
+      console.log('submit', this.validateForm.value);
+    } else {
+      Object.values(this.validateForm.controls).forEach(control => {
+        if (control.invalid) {
+          control.markAsDirty();
+          control.updateValueAndValidity({ onlySelf: true });
+        }
+      });
+    }
+  }
+  form: FormGroup;
+  isLoading: boolean = false;
+  itineraryId:string='';
+  isVisible = false;
+
+  constructor(
+
+    private fb: FormBuilder,
+    private itenaryService: ItenaryService,
+    private route: ActivatedRoute,
+    private router: Router,
+  ) {
+    this.form = this.fb.group({
+      itenaryId: ['', [Validators.required]],
+      itenaryName: ['', [Validators.required]],
+      tag: ['', [Validators.required]],
+      startTime: ['', [Validators.required]],
+      endTime: ['', [Validators.required]],
+      cost: ['', [Validators.required]],
+      startLocation: ['', [Validators.required]],
+      endLocation: ['', [Validators.required]],
+      Notes: ['', [Validators.required]],
+    });
+
+  }
+
+  ngOnInit() {
+
+    this.route.params.subscribe(params => {
+      this.itineraryId = params['itineraryId'];
+      console.log(this.itineraryId)
+    });
+
+    this.itineraryId = this.route.snapshot.paramMap.get('itineraryId')??''
+    this.itenaryService.getItenaryById(this.itineraryId).subscribe((itinerary)=>{
+      //console.log(trip[0].id);
+      this.form.controls['itenaryName'].setValue(`${itinerary[0].itineraryName}`)
```

Comment:
Assumptions everywhere. Are you sure there's at least one item on the itinerary array?